rust-toolchain.toml

```diff
@@ -1,5 +1,10 @@
 [toolchain]
-# Please update rust-version in **/Cargo.toml files when changing channel:
-channel = "1.56.0"
+# This specified the version we use to build.
+channel = "1.57.0"
+# In order to provide provide backward compatibility, we define minimum version.
+# Current minimum version is "1.56.0", please change rust-version in `**/Cargo.toml`.
+# The minimum supported version is important for outside packages, importing packages
+# from `near/nearcore`. That means we are limited to features available in `1.56.0`.
+# However, we can still benifit from compiler optimizations, improved `cargofmt`, `clippy`, etc.
```

Comment:
Can we reword this documentation to make it clear that individual crates can have a lower `rust-version` but never higher than this toolchain version to ensure it's actually compilable?
So whenever a crate's `rust-version` is bumped more than the current toolchain version, the toolchain version has to be updated too.

Reply:
I'll rewrite this section to include this information.

Reply:
@miraclx I rewrote the documentation. I think it's better now.